enhancements/helm3/namespace-scoped-helm-repository-crd.md

```diff
@@ -0,0 +1,196 @@
+---
+title: namespace-scoped-helm-repository-crd
+authors:
+  - "@zonggen"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2021-11-03
+last-updated: 2021-11-03
+status: implementable
+---
+
+# Namespace-scoped Helm Repository CRD
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Right now we support [`cluster-scoped` Helm repository CRD](https://github.com/openshift/api/blob/master/helm/v1beta1/0000_10-helm-chart-repository.crd.yaml#L17). This requires admin-level permissions when developers want to add a Helm repository in the OpenShift Console. We would like to add the ability to install Helm repository as a custom resource but with `scope: namespaced`.
+
+
+## Motivation
+
+While the current cluster-scoped CRD for Helm repository provides ability for admins to add Helm repositories as custom resources, regular users/developers need an admin's permission in order to add Helm repositories of their choice. In order to further increase the customizability of Helm within OpenShift, a new namespace-scoped CRD for Helm repositories is needed.
+
+### Goals
+
+- New namespace-scoped CRD created for Helm repositories: ProjectHelmChartRepository
+- Able to create the new namespace-scoped custom resource, ProjectHelmChartRepository
+- Document in helm section describing the difference between ProjectHelmChartRepository and HelmChartRepository
+
+### Non-Goals
+
+- Change the existing cluster-scoped CRD, HelmChartRepository
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As an OpenShift developer, I would like to be able to add Helm repositories within my project without asking for an administrator's permissions
```

Comment:
presumably only people w/ specific permission will be able to see this repository, it doesn't become globally visible, right?

What is the expected RBAC permission that will determine who can see this repository?  project viewer? project editor?  explicit role being granted to a user in that namespace?

Reply:
> presumably only people w/ specific permission will be able to see this repository, it doesn't become globally visible, right?

Yes, only users that have _at least_ read access to that that namespace/project will be able to see.